src/ptc/util/misc.clj

```diff
@@ -144,6 +144,16 @@
       (throw (IllegalArgumentException. (format "Bad GCS URL: '%s'" url))))
     [bucket (or object "")]))
 
+(defn gs-object-leaf
+  "Return object LEAF from a gs://bucket/object URL."
+  [url]
+  (-> (parse-gs-url url) second (str/split #"/") last))
+
+(defn JMS->re-prefixed-path
+  "Given a `jms-cloud-path`, re-prefix it with a new prefix."
+  [prefix jms-cloud-path]
+  (str/join "/" [prefix (gs-object-leaf jms-cloud-path)]))
```

Comment:
Again: this is just adding a leaf name to a path.
Nothing JMS about it.
"Tack leaf of `path` onto the end of `prefix`." … or something.